chore(provider): PagerDuty scopes added and fix: mongodb connection check

KanvaBhatia opened the pull request:
<!-- 
Thanks for creating this pull request 🤗

Please make sure that the pull request is limited to one type (docs, feature, etc.) and keep it as small as possible. You can open multiple prs instead of opening a huge one.
-->

<!-- If this pull request closes an issue, please mention the issue number below -->
Closes #1031 and #903

## 📑 Description
<!-- Add a brief description of the pr -->
Added `validate_scopes` for `PagerdutyProvider`. 
Included the following scopes - 
1. `incidents.read`
2. `incidents.write`
3. `webhook_subscriptions.read`
4. `webhook_subscriptions.write`
<!-- You can also choose to add a list of changes and if they have been completed or not by using the markdown to-do list syntax
- [ ] Not Completed
- [x] Completed
-->
- [x] Scopes added and created `validate_scopes`
- [x] Edit Documentation - Add Scopes, Edit Inputs and Authentication Parameters

Also fixed mongodb always validating scopes.

## ✅ Checks
<!-- Make sure your pr passes the CI checks and do check the following fields as needed - -->
- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [x] All the tests have passed

## ℹ Additional Information
<!-- Any additional information like breaking changes, dependencies added, screenshots, comparisons between new and old behavior, etc. -->
Pagerduty - 
If api key is wrong (Unauthorized) - 
![image](https://github.com/keephq/keep/assets/56185464/e9628c63-6f58-48cc-ae8e-98a55c96ef6d)

If valid api key and authorized - 
![image](https://github.com/keephq/keep/assets/56185464/989bb162-8921-48ad-8cc7-f4a76f363873)


Mongodb - 
If wrong URI - 
![image](https://github.com/keephq/keep/assets/56185464/f1874b81-4a68-4acc-8a11-094d573c445f)

If right URI -
![image](https://github.com/keephq/keep/assets/56185464/217a053e-c0f2-4747-b1fe-380b8ff31073)

talboren commented on keep/providers/pagerduty_provider/pagerduty_provider.py:
```diff
@@ -71,6 +99,37 @@ def validate_config(self):
+                    scopes[scope.name] = True
+                else:
+                    scopes[scope.name] = response.reason
+            except Exception as e:
```
```suggestion
            except Exception:
```

talboren commented on keep/providers/pagerduty_provider/pagerduty_provider.py:
```diff
@@ -71,6 +99,37 @@ def validate_config(self):
+                else:
+                    scopes[scope.name] = response.reason
+            except Exception as e:
+                self.logger.error("Error validating scopes", extra={"error": str(e)})
```
```suggestion
                self.logger.exception("Error validating scopes")
```

talboren reviewed (changes requested):
Amazing, thanks! 👑 two small comments and linter yelling about some unused variable and we’re set!

KanvaBhatia commented:
Hey @talboren I by mistake pushed the mongodb changes also in the same branch and they became part of the same PR.
Added the details of that fix in the initial comment.
Fixed the unused variable `e` in mongodb provider and also `logger.exception` instead of `logger.error`.

talboren commented:
> Hey @talboren I by mistake pushed the mongodb changes also in the same branch and they became part of the same PR. Added the details of that fix in the initial comment. Fixed the unused variable `e` in mongodb provider and also `logger.exception` instead of `logger.error`.

No worries! Looks great.

talboren reviewed (approved):
LGTM
